designsafe/apps/projects/managers/publication.py

```diff
@@ -178,6 +179,72 @@ def draft_publication(
         )
     return responses
 
+def amend_publication(project_id, revision=None):
+    """Amend a Publication
+    
+    Update Amendable fields on a publication and the corrosponding DataCite
+    records. These changes do not produce a new version of a publication, but
+    they do allow for limited changes to a published project. This is currently
+    configured to support "Other" publications only.
+    
+    :param str project_id: Project uuid to amend
+    :param int revision: Revision number to amend
+    """
+    mgr = ProjectsManager(service_account())
+    prj = mgr.get_project_by_id(project_id)
+    pub = BaseESPublication(project_id=project_id, revision=revision)
+    if prj.project_type != 'other':
+        return
+
+    prj_dict = prj.to_body_dict()
+    pub_dict = pub.to_dict()
+    _delete_unused_fields(prj_dict)
+
+    # weird key swap for old issues with awardnumber(s)
+    award_number = prj.award_number or []
+    if not isinstance(award_number, list):
+        award_number = []
+    prj_dict['value']['awardNumbers'] = award_number
+    prj_dict['value'].pop('awardNumber', None)
+
+    amends_dict = {
+        'nhTypes': [],
+        'dataType': '',
+        'awardNumbers': [],
+        'associatedProjects': [],
+        'keywords': '',
+        'description': '',
+    }
+    for key in amends_dict:
+        amends_dict[key] = prj_dict['value'][key]
+
+    pub_dict['project']['value'].update(amends_dict)
+    pub.update(**pub_dict)
+
+    return pub
+
+def amend_datacite_doi(publication):
```

Comment:
Should we make this method prod-only so that we can test it in dev/staging without making real datacite calls?

Reply:
That's a good idea, I could set it up to use the test API that DataCite has, but we'd need to change some environment variables to include that. If I check for the `DESIGNSAFE_ENVIRONMENT` we could switch the  `DATACITE_USER` to a  `TEST_DATACITE_USER`... I'm not too familiar with how that looks on the staging/prod side of things though.
https://github.com/DesignSafe-CI/portal/blob/master/designsafe/apps/projects/managers/datacite.py#L16

Reply:
If we're confident in how we're formatting the calls then the best practice might be to return early in non-prod environments, but include good unit tests. I'd be happy to submit a supplemental PR to this one with test coverage.

Reply:
Okay, I'll just have it return early if we're not in the prod environment.